src/core/dash_internal.h

```diff
@@ -337,32 +341,27 @@ template <typename _Key, typename _Value, typename Policy = DefaultSegmentPolicy
       std::swap(value[slot_a], value[slot_b]);
     }
 
-    // calls for each busy slot: cb(iterator, probe)
-    template <typename Cb> void ForEachSlot(Cb&& cb) const {
+    template <typename This, typename Cb> void ForEachSlotImpl(This obj, Cb&& cb) const {
```

Comment:
1. curious, why you added This template? 
2. You also use `this` at line 345-346. is it intended?